accepted/future-releases/extension-types/feature-specification.md

```diff
@@ -766,6 +766,8 @@ the treatment of expressions like `e as V`.*
 
 A compile-time error occurs if a type parameter of an extension type
 declaration occurs in a non-covariant position in the representation type.
+A compile-time error occurs if the representation type of an extension type
+declaration is a bottom type.
```

Comment:
That's more than what I was asking for, but probably not too much. It's not losing any actually useful functionality.

Have to make sure that the phrasing refers to the (type-alias-expansion of) the declared representation type, and not the actual representation type after type parameter instantiation of the extension type.

That is, `extension type E<T>(T v) { ... }` is valid, since `T` is not a bottom type, and `E<Never>` is valid because it's going to be impossible to prevent anyway. 

Which also means that if you really, really want a `Never` represenation type, you can do:
```dart
extension type _E<T>(T it) { ... }
typedef E = _E<Never>;
```
so we're not  *preventing* a bottom representation type, we're just preventing the extension type declaration itself from *knowing* (and using) that the type is a bottom type.

(That was what I was trying to achieve in #3383 by disallowing implementing interface types when the representation type is a bottom type, but for safety, it should probably disallow implementing extension types as well, at which point we might as well disallow the bottom representation type entirely. It's easier to explain.)

Reply:
Added commentary to explain this. No changes to normative text was needed.